Sources/KVKCalendar/WeekData.swift

```diff
@@ -10,23 +10,19 @@
 import Foundation
 
 final class WeekData: EventDateProtocol, ScrollableWeekProtocol {
-    let days: [Day]
+    var days: [Day]
     var date: Date
     var events: [Event] = []
     var recurringEvents: [Event] = []
     var daysBySection: [[Day]] = []
     
     init(data: CalendarData, startDay: StartDayType, maxDays: Int) {
         self.date = data.date
-        var item = startDay
-        if maxDays != 7 {
-            item = .sunday
-        }
-        var tempDays = data.months.reduce([], { $0 + $1.days })
-        let startIdx = tempDays.count > maxDays ? tempDays.count - maxDays : tempDays.count
-        let endWeek = data.addEndEmptyDays(Array(tempDays[startIdx..<tempDays.count]), startDay: item)
-        tempDays.removeSubrange(startIdx..<tempDays.count)
-        self.days = data.addStartEmptyDays(tempDays, startDay: item) + endWeek
+        self.days = WeekData.getDates(
```

Comment:
why static func?

Reply:
Since I need to set the array of dates for `self.dates` in initialization, I figured I need to use a static function, otherwise, I can not reuse the function inside `updateDaysBySection`. Would you recommend another way?

Reply:
try to this `var days: [Day] = []`... create an empty array

Reply:
and I think you can use the function `self.days = getDates()`